### PR TITLE
New Xero Bank Feeds integration

### PR DESCRIPTION
**Pull request by d-laing**
# Description

Adds the documentation for the new Xero Bank Feeds integration.

## Type of change

- [x] New document(s)/updating existing

## Reviews and merging

You are responsible for getting your PR merged. Address review comments promptly and **make sure to merge the PR when ready**.
Feel free to 'Enable automerge' - your PR will automatically merge when accepted and passing the build.

### CODE REVIEW

File: docs/bank-feeds-api/overview.mdx
Comment (d-laing) [inline docs/bank-feeds-api/overview.mdx:1]: @mcclowes - I also need to update the integration cards component on this page.

File: docs/bank-feeds-api/xero-bank-feeds/xero-bank-feeds.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+1. To start, create a Xero app in the <a href="https://developer.xero.com/" target="_blank">Xero developer portal</a>.
+
+   :::caution Approval required
+   You'll need to obtain Xero's approval for Bank Feeds.
+   :::
+
+2. [Set up the integration](/bank-feeds-api/xero-bank-feeds/xero-bank-feeds-setup).
+
+3. Your SMB users set up their bank feeds. They can start this process directly from your app's UI, or use the "Add bank account" feature in Xero to search for the name of your institution and then launch your UI. See [SMB user account connection](/bank-feeds-api/xero-bank-feeds/xero-bank-feeds-smb-user).
```
Comment (d-laing) [inline docs/bank-feeds-api/xero-bank-feeds/xero-bank-feeds.md:32]: @DaveHCodat - Are you sure this should say "launch your UI"? Does it not launch the Codat UI?

File: docs/bank-feeds-api/overview.mdx [hunk 2]
```diff
@@ -143,7 +143,9 @@ Here's a high-level overview of the setup and connection flow for Bank Feeds API
 
    - For Sage Bank Feeds, you surface the ID of the data connection you created in your application. The SMB user must then enter this ID in the integration's authorization UI.
 
-6. The SMB user follows the instructions in the integration's authorization UI. They choose which source bank account they want to connect to their accounting platform.
+  - For QBO Bank Feeds, you redirect the user to the `linkUrl` returned in the data connection response to load the Codat-provided account mapping UI.
```
Comment (pzaichkina) [inline docs/bank-feeds-api/overview.mdx:146]: Isn't this now a duplication of line 142?

File: docs/bank-feeds-api/xero-bank-feeds/xero-bank-feeds-push-bank-transactions.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+:::note Pushing historic transactions
+You can push bank transactions to Xero which are dated up to one year prior to the current date. Pushes containing bank transactions dated older than one year will fail.
+
+The `date` of a historic transaction must be later than the `feedStartDate` on the source bank account, which is determined by the **Feed start date** selected by the SMB user in the account mapping UI.
+:::
+
+## Push bank transactions to Xero
+
+To push bank transactions for a `connected` source bank account, make the following requests to the Codat API. All push requests are asynchronous. Bank feeds are sent to Xero immediately, not on a schedule.
```
Comment (Jbloemhoff) [inline docs/bank-feeds-api/xero-bank-feeds/xero-bank-feeds-push-bank-transactions.md:78]: Would change "Bank feeds are sent to Xero immediately" to "Bank feeds transactions" are sent to Xero immediately

File: docs/bank-feeds-api/xero-bank-feeds/xero-bank-feeds-push-bank-transactions.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
+---
+title: "Push bank transactions to Xero"
+sidebar_label: Push bank transactions to Xero
```
Comment (pzaichkina) [inline docs/bank-feeds-api/xero-bank-feeds/xero-bank-feeds-push-bank-transactions.md:3]:
```suggestion
sidebar_label: Push bank transactions
```

File: docs/bank-feeds-api/xero-bank-feeds/xero-bank-feeds.md (hunk 1 above)
Comment (DaveHCodat) [reply docs/bank-feeds-api/xero-bank-feeds/xero-bank-feeds.md:32]: Yes this is correct, as Xero first redirect to client UI to authenticate the user

File: docs/bank-feeds-api/xero-bank-feeds/xero-bank-feeds-push-bank-transactions.md [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+- View the details of source bank accounts, including their connection statuses.
+- Push bank transactions to a target bank account in Xero.
+
+## Prerequisites
+
+- You've [set up the Xero Bank Feeds integration](/bank-feeds-api/xero-bank-feeds/xero-bank-feeds-setup).
+
+## View bank account details
+
+Call  [GET connectionInfo/bankFeedAccounts](/bank-feeds-api#/operations/get-bank-feeds) to view details of the source bank accounts for a specified company and data connection.
```
Comment (pzaichkina) [inline docs/bank-feeds-api/xero-bank-feeds/xero-bank-feeds-push-bank-transactions.md:24]: the endpoints are not referenced uniformly in the article. 
this one gives the path and doesn't say "endpoint", but in line 80 you use a readable name and the word "endpoint".
image

File: docs/bank-feeds-api/xero-bank-feeds/xero-bank-feeds-setup.md [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+
+- Create a company and a data connection to Xero.
+- Provide Codat with source bank accounts you want to make available for your SMB users to connect to.
+- Redirect the user to the Codat-provided account mapping UI, where they can connect their source bank accounts to target bank accounts in Xero.
+
+## Prerequisites
+
+Before setting up the integration, make sure that:
+
+- You've [joined the Xero App Partner Program](/accounting/xero/xero-app-partner-program).
```
Comment (pzaichkina) [inline docs/bank-feeds-api/xero-bank-feeds/xero-bank-feeds-setup.md:19]:
```suggestion
- You've [joined the Xero App Partner Program](/integrations/accounting/xero/xero-app-partner-program).
```

File: docs/bank-feeds-api/xero-bank-feeds/xero-bank-feeds-setup.md [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+   - [Retrieve your app's secure keys](/integrations/accounting/xero/accounting-xero-setup#retrieve-your-apps-secure-keys) and then add them to the integration.
+   - [Enable the Xero integration](/integrations/accounting/xero/accounting-xero-setup#enable-the-xero-integration).
+
+   Bank feeds functionality is part of our existing Xero accounting integration. To configure the integration to access bank feeds, use the same Redirect URI and ensure the **Enable bank feeds** toggle is selected.
+   
+- Xero have enabled the _Xero Bank Feeds API_ for your registered app.
+
+### Create a company and data connection, then add bank accounts​
+
+1. Using [POST /companies](/codat-api#/operations/create-company), create a company to represent your SMB user:
```
Comment (pzaichkina) [inline docs/bank-feeds-api/xero-bank-feeds/xero-bank-feeds-setup.md:31]: I guess same question about referencing the endpoints.

File: docs/bank-feeds-api/xero-bank-feeds/xero-bank-feeds-setup.md [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+   POST https://api.codat.io/companies/<COMPANY_ID>/connections
+   ```
+   
+   ```json title="Request body - Xero"
+   {
+     "platformKey": "gbol"
+   } 
+   ```
+   
+   The endpoint returns a `dataConnection` object, in `PendingAuth` status, containing a connection `id` and a `linkUrl`. Later, you'll use the `linkUrl` to redirect the SMB user to the account mapping UI.
```
Comment (pzaichkina) [inline docs/bank-feeds-api/xero-bank-feeds/xero-bank-feeds-setup.md:57]:
```suggestion
   The endpoint returns a `dataConnection` object in `PendingAuth` status, containing a connection `id` and a `linkUrl`. Later, you'll use the `linkUrl` to redirect the SMB user to the account mapping UI.
```

File: docs/bank-feeds-api/xero-bank-feeds/xero-bank-feeds-setup.md [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+   } 
+   ```
+
+3. Using [PUT / bankFeedAccounts](/bank-feeds-api#/operations/create-bank-feed), add one or more source bank accounts:
+   
+   ```http title="Create bank feed bank accounts"
+   PUT /companies/<COMPANY_ID>/connections/<CONNECTION_ID>/connectionInfo/bankFeedAccounts
+   ```
+   
+   For the `<CONNECTION_ID>`, use the connection `id` in the response from `POST /companies`.
```
Comment (pzaichkina) [inline docs/bank-feeds-api/xero-bank-feeds/xero-bank-feeds-setup.md:79]:
```suggestion
   For the `<CONNECTION_ID>`, use the connection `id` you received in the response from `POST /companies`.
```

File: sidebars/bank-feeds-api.js
Comment (pzaichkina) [inline sidebars/bank-feeds-api.js:1]: I am not sure which exactly sidebar file / change may have led to this, but if you navigate to Integrations -> Bank feeds -> Overview, it leads to a page with no sidebar on it - /integrations/bankfeeds/overview

Is this page no longer needed and should be totally removed, or should it be part of the overall bank feeds nav?

File: docs/bank-feeds-api/overview.mdx
Comment (mcclowes) [reply docs/bank-feeds-api/overview.mdx:1]: I can show you how if you put time in

File: docs/bank-feeds-api/overview.mdx (hunk 2 above)
Comment (d-laing) [reply docs/bank-feeds-api/overview.mdx:146]: There is now a combined step: "For QBO Bank Feeds and Xero Bank Feeds, you redirect the user to the `linkUrl` returned in the data connection response to load a Codat-provided UI."

File: docs/bank-feeds-api/xero-bank-feeds/xero-bank-feeds-push-bank-transactions.md (hunk 4 above)
Comment (d-laing) [reply docs/bank-feeds-api/xero-bank-feeds/xero-bank-feeds-push-bank-transactions.md:3]: I will also align QBO Bank Feeds to use this format.

**Review by DaveHCodat — APPROVED** [review]

File: docs/bank-feeds-api/xero-bank-feeds/xero-bank-feeds-push-bank-transactions.md (hunk 5 above)
Comment (d-laing) [reply docs/bank-feeds-api/xero-bank-feeds/xero-bank-feeds-push-bank-transactions.md:24]: I clearly couldn't make up my mind! :)

I have changed all the requests to say "endpoint" and updated the push example above to `POST /push/bankAccounts/<ACCOUNT_ID>/bankTransactions`. 

One to take away and consider for the style guide.

File: docs/bank-feeds-api/xero-bank-feeds/xero-bank-feeds-setup.md (hunk 7 above)
Comment (d-laing) [reply docs/bank-feeds-api/xero-bank-feeds/xero-bank-feeds-setup.md:31]: Left as-is.

**Review by Jbloemhoff — APPROVED** [review]

**Review by pzaichkina — APPROVED** [review]

**Pull request merged**